doc/source/groups_of_atoms.rst

```diff
@@ -8,79 +8,79 @@ MDAnalysis has a hierarchy of :class:`~MDAnalysis.core.groups.Atom` containers t
-    >>> a.residue.segment.residues
-    <ResidueGroup with 214 residues>
 
 For information on adding custom Residues or Segments, have a look at :ref:`adding-residue-label`.
 
 ---------------------------
 Fragments
 ---------------------------
 
-Certain analysis methods in MDAnalysis also make use of additional ways to group atoms. A key concept is a fragment. A fragment is what is typically considered a molecule: a group of atoms where each atom is bonded to at least one other atom in the fragment, and are not bonded to any atoms outside the fragment. (A 'molecule' in MDAnalysis methods :ref:`refers to a GROMACS-specific concept <molecule-label>`). The fragments of a Universe are determined by MDAnalysis as a derived quantity. They can only be determined if bond information is available.
+Certain analysis methods in MDAnalysis also make use of additional ways to group atoms. A key concept is a fragment. A fragment is what is typically considered a molecule: an AtomGroup where any atom is reachable from any other atom in the AtomGroup by traversing bonds, and none of its atoms is bonded to any atoms outside the AtomGroup. (A 'molecule' in MDAnalysis methods :ref:`refers to a GROMACS-specific concept <molecule-label>`). The fragments of a Universe are determined by MDAnalysis as a derived quantity. They can only be determined if bond information is available.
+
+The fragments of an :class:`~MDAnalysis.core.groups.AtomGroup` are accessible via the :attr:`fragments` property. Below is a Universe from a GROMACS TPR file of lysozyme (`PDB ID: 2LYZ <http://www.rcsb.org/structure/2LYZ>`_) with 101 water molecules. While it has 230 residues, there are only 102 fragments: 1 protein and 101 water fragments.
+
+.. ipython:: python
+
+    len(u.residues)
+    len(u.atoms.fragments)
+
 
-The fragments of an :class:`~MDAnalysis.core.groups.AtomGroup` are accessible via the :attr:`fragments` property. Below is a Universe from a GROMACS TPR file of lysozyme (`PDB ID: 2LYZ <http://www.rcsb.org/structure/2LYZ>`_) with 101 water molecules. While it has 2263 residues, there are only 102 fragments: 1 protein and 101 water fragments. ::
+See :ref:`topology-objects-label` for more on bonds and which file formats give MDAnalysis bond information.
 
-    >>> import MDAnalysis as mda
-    >>> from MDAnalysis.tests.datafiles import TPR2019B3
-    >>> u = mda.Universe(TPR209B3)
-    >>> len(u.residues)
-    2263
-    >>> u.residues.resnames
-    array(['LYSH', 'VAL', 'PHE', ..., 'SOL', 'SOL', 'SOL'], dtype=object)
-    >>> u.atoms.fragments
-    (<AtomGroup with 1960 atoms>, <AtomGroup with 3 atoms>, <AtomGroup with 3 atoms>, ..., <AtomGroup with 3 atoms>, <AtomGroup with 3 atoms>, <AtomGroup with 3 atoms>)
-    >>> len(u.atoms.fragments)
-    102
+You can also look at which fragment a particular :class:`~MDAnalysis.core.groups.Atom` belongs to:
 
-See :ref:`topology-objects` for more on bonds and which file formats give MDAnalysis bond information.
+.. ipython:: python
```

Comment:
The syntax highlighting seems to be broken until this block. I can’t see how it’s different to preceding ones though

Reply:
This is annoying and [seems low priority](https://github.com/ipython/ipython/issues/11940). The only thing that correlated to highlight breaking is block length (maybe a maximum of 1 output?); separately, each In/Out statement is highlighted. The IPython directive *is* still in beta -- I'll break up blocks or switch to normal code blocks for now.